feat!: Reworks and Quality of Life updates

ShadowCat117 opened the pull request:
See https://github.com/Wynntils/Wynntils/milestone/2 for what is planned to be implemented

ShadowCat117 committed:
feat: Display tooltip for overlays with parent disabled (#3237)

ShadowCat117 committed:
feat: Paginate /functions list (#3245)

kristofbolyai committed:
Merge branch 'main' into reworks-&-qol

kristofbolyai committed:
fix: Make location sharing round coordinates the same way functions do (#3250)

ShadowCat117 committed:
feat: Use a custom font for rendering Wynntils marker on nametags (#3264)

ShadowCat117 committed:
feat: Add autocomplete to /switch and /server info (#3263)

ShadowCat117 committed:
feat: Add all alignment options to unid revealed icon location (#3260)

ShadowCat117 committed:
feat: Add option to move tooltip position for cosmetics in scrap menu (#3257)

* fix: Fix highlight duplicate cosmetics highlight rendering

* feat: Add option to move tooltip position for cosmetics in scrap menu

ShadowCat117 committed:
feat: Add 2 new item highlight texture options (#3255)

ShadowCat117 committed:
feat: Add overlay position, size and alignment buttons to configurable screens (#3254)

* feat: Add overlay position and size buttons

* fix: Fix scroll offset not persisting between screens

* refactor: Make overlay configs non hidden

ShadowCat117 committed:
feat: Add American spelling for defence stat filters (#3253)

feat: Add American spelling for defence stat silters

magicus commented:
Sounds like this should be a `feat!` when it is integrated. You're doing a massive amount of work here, @ShadowCat117, thanks!

ShadowCat117 committed:
feat: Add accessory quick filters to item guide, reposition widgets and new keybinds (#3265)

* feat: Add accessory quick filters to item guide, reposition widgets

* feat: Add keybinds to open, charm, tome and aspect guide

ShadowCat117 committed:
feat: Use rainbow effect for perfect items (#3258)

* feat: Use rainbow effect for perfect items

* chore: Rename parameter

ShadowCat117 committed:
Merge branch 'main' into reworks-&-qol

ShadowCat117 committed:
feat: Add grouping, sorting and limit options to bomb bell overlay (#3262)

* feat: Add limit option to bomb bell overlay

* fix: Fix description

* feat: Add grouping and sort configs, change default max

* refactor: Update render tasks on tick

ShadowCat117 committed:
fix: Properly port item record to 2.1 and Minecraft 1.21.4 (#3256)

* fix: Fix item record rendering

* fix: Rework saved items to use registered item name and custom model data

* refactor: Use ItemStack codec

ShadowCat117 committed:
feat: Add update screen before join (#3252)

* refactor: Use non deprecated method for mc version check

* chore: Remove TODO for update streams config

* fix: Fix typo

* feat: Add update screen before join

* fix: Check there is mod update info on join

* refactor: Handle should prompt logic in service

* refactor: Handle download progress in FileUtils

ShadowCat117 committed:
feat: Add texture options to universal bars (#3275)

ShadowCat117 committed:
Merge branch 'main' into reworks-&-qol

ShadowCat117 committed:
feat: Various minor UI improvements (#3276)

ShadowCat117 committed:
Merge branch 'main' into reworks-&-qol

kristofbolyai committed:
ci: Autopromote 'release' tagged PRs and allow promoting by hand (#3283)

Now, we don't need to add [skip ci] on every PR, rather, we can add a tag to the PR that we want to make a release with. This way, we don't need to filter out [release] from the PR title, and tags are much more visible and filterable. I also separated beta and release scripts to allow us to run them by hand separately.

ShadowCat117 committed:
fix: Don't throw exception on failing to deserialize SavedItem (#3285)

ShadowCat117 committed:
feat: Rework custom content book (#3244)

* feat: Rework custom content book

* refactor: Use a string list for rewards in content book dump feature

* chore: Add back display overall progress

* feat: Improvements to tracked/hovered widget

* chore: Move player progress widget

* fix: Fix widgets staying bold after unhovering

* feat: Change tutorial highlight, remove tracked quest highlight

* feat: Recolour description

* feat: Shrink wynntils menu tag

* feat: Add upfixer for replace config

* ci: spotless formatting

* refactor: Create public ShiftBehavior enum

* refactor: Use EnumUtils.toJsonFormat in upfixer

* feat: Rework overall progress parsing to not trigger every open of Wynntils menu

* feat: Add search bar texture

* feat: Respect shift config when left clicking

* fix: Don't recenter mouse when opening Wynntils menu from content book

---------

Co-authored-by: ShadowCat117 <8337467+ShadowCat117@users.noreply.github.com>

ShadowCat117 committed:
feat: Add color template to universal bars (#3287)

* fix: Fix commander bar rendering

* feat: Add from_hex function

* feat: Add color template to universal bars

* refactor: Calculate colour on tick

* refactor: Improve readability of upfixer

ShadowCat117 committed:
ci: Rework release workflows to use manually set changelog (#3289)

ShadowCat117 committed:
feat: Improve changelog (#3288)

* feat: Improve changelog

* chore: Remove unused UrlIds

* refactor: Introduce ChangelogMap

* ci: spotless formatting

---------

Co-authored-by: ShadowCat117 <8337467+ShadowCat117@users.noreply.github.com>

ShadowCat117 committed:
feat: Add Wynntils Hint Messages (#3284)

* feat: Add Wynntils Info Messages

* chore: Rename feature

* chore: Update feature description

* fix: Handle unbound keys when not using English

* refactor: Introduce HintAction enum

* refactor: Refactor formatHint

* feat: Allow for custom formatting in hints

* refactor: Use StyledText

* chore: Update md5

ShadowCat117 committed:
Merge branch 'main' into reworks-&-qol
